### PR TITLE
Adding new indexes for unindexed JSON search queries

### PR DESCRIPTION
**Pull request by sumansreenivasan**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/resources/db/changelog/db.changelog-RDM-8463.xsl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+
+    <changeSet id="rdm-8463" author="suman.sree@hmcts.net">
```
Comment (mario-paniccia) [inline src/main/resources/db/changelog/db.changelog-RDM-8463.xsl:6]: you need to add the changeset to : 
https://github.com/hmcts/ccd-data-store-api/blob/fc07f8b55fb67379b2e7db457fa08d347d004c7f/src/main/resources/db/changelog/db.changelog-master.xml#L1
for it to be executed

File: src/main/resources/db/changelog/db.changelog-RDM-8463.xsl [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        <sql dbms="postgresql"
+             endDelimiter="\nGO"
+             splitStatements="true"
+             stripComments="true">
+            CREATE INDEX  idx_case_data_appeal_appellant_identity_nino ON public.case_data USING BTREE ((TRIM(UPPER(data#>>'{appeal,appellant,identity,nino}'))));
```
Comment (mario-paniccia) [inline src/main/resources/db/changelog/db.changelog-RDM-8463.xsl:11]: I'm a bit confused didn't Sateesh added this index as part of:
https://github.com/hmcts/ccd-data-store-api/pull/939/files

File: src/main/resources/db/changelog/db.changelog-RDM-8463.xsl [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            CREATE INDEX  idx_case_data_deceaseddateofbirth ON public.case_data USING BTREE ((TRIM(UPPER(data#>>'{deceasedDateofBirth}'))));
+            CREATE INDEX  idx_case_data_createdingapsfrom_dwpstate ON public.case_data USING BTREE ((TRIM(UPPER(data#>>'{createdingapsfrom,dwpstate}'))));
+            CREATE INDEX  idx_case_data_d8petitionerlastname ON public.case_data USING BTREE ((TRIM(UPPER(data#>>’{D8PetitionerLastname}'))));
```
Comment (smathangi) [inline src/main/resources/db/changelog/db.changelog-RDM-8463.xsl:14]: These indexes are already applied to prod manually. So we need to use "Create if not exists" logic to handle that

File: src/main/resources/db/changelog/db.changelog-RDM-8463.xsl [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (mario-paniccia) [inline src/main/resources/db/changelog/db.changelog-RDM-8463.xsl:2]: the ticket is requesting to add indexes for all the unindexed JSON search currently on PRod. I can only see we are adding the liquibase scripts for the 4 indexes that were introduced manually on Prod last week. But, are there other indexes that we still need to add? Did we ask Alliu?

File: src/main/resources/db/changelog/db.changelog-RDM-8463.xsl (hunk 2 above)
Comment (mario-paniccia) [reply src/main/resources/db/changelog/db.changelog-RDM-8463.xsl:11]: I've closed that one because it's duplicate with this PR

**Comment by sumansreenivasan** [comment]
This PR need to me added as a part of RDM-8485

**Pull request closed without merging**